halo/evmengine/keeper/abci_internal_test.go

```diff
@@ -162,11 +166,11 @@ func TestKeeper_PrepareProposal(t *testing.T) {
 		require.NoError(t, err)
 
 		// build next two blocks and get the PayloadID of the second
-		mockEngine.pushPayload(t, ctx, ap.LocalAddress(), latestBlock.Hash(), ts)
-		nextBlock, blockPayload := mockEngine.nextBlock(t, latestHeight+1, uint64(ts.Unix()), latestBlock.Hash(), ap.LocalAddress())
+		mockEngine.pushPayload(t, ctx, keeper.addrProvider.LocalAddress(), latestBlock.Hash(), ts)
```

Comment:
rather use `ap` above, keep you grubby paws out of the keeper internals. Just because this is a internal test, doesn't mean we can disrespect good hygiene 😉